engine/common/follower/engine.go

```diff
@@ -80,7 +80,12 @@ func New(
 		tracer:         tracer,
 	}
 
-	con, err := net.Register(network.ReceiveBlocks, e)
+	channel := network.ReceiveBlocks
+	if !me.StakedNode() {
+		channel = network.PublicPushBlocks
+	}
+
+	con, err := net.Register(channel, e)
```

Comment:
Could we maybe instead just pass the channel into the engine as an argument? This way we can get rid of the `StakedNode()` function

Reply:
An easier way to do it is to define a function option, which we can use to change this channel from the default. This way we won't need to modify all node `main.go` files, and only need to change the Observer file